backend/plugin/db/util/ssl.go

```diff
@@ -15,38 +15,47 @@ func GetTLSConfig(ds *storepb.DataSource) (*tls.Config, error) {
 	if !ds.GetUseSsl() {
 		return nil, nil
 	}
+
+	cfg := &tls.Config{}
+
+	// Handle client certificates for mutual TLS authentication
+	// Client certificates can be used with or without server verification
+	if err := configureClientCertificates(ds, cfg); err != nil {
+		return nil, err
+	}
+
+	// Handle server certificate verification
+	if !ds.GetVerifyTlsCertificate() {
+		// Certificate verification is disabled (default for backward compatibility)
+		// This accepts any certificate presented by the server but still uses encryption
+		cfg.InsecureSkipVerify = true
+		return cfg, nil
```

Comment:
I see two places setting InsecureSkipVerify = true. This will return early if !VerifyTlsCertificate which ignores intermediate certificates. What's the expected behavior?